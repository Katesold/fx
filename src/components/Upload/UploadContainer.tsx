import { DragAndDrop } from './DragAndDrop';
import { StyledUploadContainer } from './styles';

export const UploadContainer = () => {
    return (
        <>
            <StyledUploadContainer>
                <DragAndDrop />
            </StyledUploadContainer>
        </>
    )
}