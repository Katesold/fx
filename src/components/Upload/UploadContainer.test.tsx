import { UploadContainer } from './UploadContainer';
import { screen, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('UploadContainer components are being tested', () => {
    const props = {
        uploadPayments: jest.fn()
    };
    it('should match snapshot', () => {
        const { container } = render(<UploadContainer {...props} />);
        expect(container).toMatchSnapshot();
        expect(screen.getByText('Drag & Drop')).toBeInTheDocument();
    });
    it('should accept file upload', async () => {
        render(<UploadContainer {...props} />);
        const file = new File(["(⌐□_□)"], "test.csv", { type: "text/csv" });
        const inputEl = document.querySelector('input') as HTMLInputElement;
        await waitFor(() => {
          userEvent.upload(inputEl, file);
        });
        expect(inputEl.files).toHaveLength(1);
    })
});