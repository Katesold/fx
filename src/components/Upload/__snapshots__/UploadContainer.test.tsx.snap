// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadContainer components are being tested should match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe buZDGq"
  >
    <h1>
      Upload Payments
    </h1>
    <div
      class="sc-gswNZR igKVwO dropzone"
      role="presentation"
      tabindex="0"
    >
      <input
        accept="text/csv,.csv,.xls,.xlsx"
        multiple=""
        style="display: none;"
        tabindex="-1"
        type="file"
      />
      <img
        alt="cloud"
        src="cloud.png"
      />
      <h4>
        Drag & Drop
      </h4>
      <p
        class="sc-dkrFOg fUSOcT"
      >
        Supported file types: 
        <span>
          .csv .xls .xlsx
        </span>
      </p>
    </div>
  </div>
</div>
`;
